src/test/java/org/broadinstitute/consent/http/db/LibraryCardDAOTest.java

```diff
@@ -0,0 +1,121 @@
+package org.broadinstitute.consent.http.db;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Date;
+import java.util.List;
+
+import com.google.gson.Gson;
+
+import org.postgresql.util.PSQLException;
+import org.postgresql.util.PSQLState;
+import org.apache.commons.lang3.RandomUtils;
+import org.broadinstitute.consent.http.models.LibraryCard;
+import org.junit.Test;
+
+public class LibraryCardDAOTest extends DAOTestHelper {
+
+  @Test
+  public void testInsertLibraryCard() {
+    LibraryCard card = createLibraryCard();
+    List<LibraryCard> all = libraryCardDAO.findAllLibraryCards();
+    assertTrue(all.contains(card));
+  }
+
+  @Test
+  public void testInsertLibraryCardNegative() {
+    Integer userId = createUser().getDacUserId();
+    Integer institutionId = createInstitution().getId();
+    String stringValue = "value";
+    try { 
+        libraryCardDAO.insertLibraryCard(0, institutionId, stringValue, stringValue, stringValue, userId, new Date());
+    } catch (Exception e) {
+        assertEquals(PSQLState.FOREIGN_KEY_VIOLATION.getState(), ((PSQLException)e.getCause()).getSQLState());
+    }
+    try { 
+        libraryCardDAO.insertLibraryCard(userId, 0, stringValue, stringValue, stringValue, userId, new Date());
+    } catch (Exception e) {
+        assertEquals(PSQLState.FOREIGN_KEY_VIOLATION.getState(), ((PSQLException)e.getCause()).getSQLState());
+    }
+    try { 
+        libraryCardDAO.insertLibraryCard(userId, institutionId, stringValue, stringValue, stringValue, 0, new Date());
+    } catch (Exception e) {
+        assertEquals(PSQLState.FOREIGN_KEY_VIOLATION.getState(), ((PSQLException)e.getCause()).getSQLState());
+    }
+  }
+
+  @Test
+  public void testUpdateLibraryCardById() {
```

Comment:
this test currently fails with the commented out lines uncommented. It is only updating some information. Specifically, this is what the test returns as the updated library card:
{"id":2,"userId":5,"institutionId":4,"eraCommonsId":"New Value","createDate":"May 10, 2021, 3:18:40 PM","updateDate":"May 10, 2021, 3:18:40 PM"}
meaning that name, email, and update user are completely ignored, even when the name is set (on line 55) prior to the update the library card returned does not include the old or new value instead it is null. Ideas as to why this is happening now and wasn't happening before? the only functional changes I made was using the row reducer instead of the row mapper

Reply:
Ah that makes sense thank you! I'll make that change